File: HackerRank Solutions/Interview Preparation Kit/Stacks and Queues/Queues- A Tale of Two Stacks.py
class MyQueue(object):
    def __init__(self):
        self.stack1 = []
        self.stack2 = []
    
    def peek(self):
        if len(self.stack1) <= 0 and len(self.stack2) <= 0:
            return -1
        if len(self.stack2) <= 0:
            while len(self.stack1) > 0:
                x = self.stack1.pop()
                self.stack2.append(x)
        return self.stack2[-1]
        
    def pop(self):
        if len(self.stack1) <= 0 and len(self.stack2) <= 0:
            return -1
        if len(self.stack2) <= 0:
            while len(self.stack1) > 0:
                x = self.stack1.pop()
                self.stack2.append(x)
        return self.stack2.pop()
        
    def put(self, value):
        self.stack1.append(value)
        

queue = MyQueue()
t = int(input())
for line in range(t):
    values = map(int, input().split())
    values = list(values)
    if values[0] == 1:
        queue.put(values[1])        
    elif values[0] == 2:
        queue.pop()
    else:
        print(queue.peek())

File: HackerRank Solutions/Interview Preparation Kit/Stacks and Queues/Queues- A Tale of Two Stacks.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static class MyQueue<T> {

        private Stack stack1;
        private Stack stack2;

        public <T> MyQueue() {
            this.stack1 = new Stack<>();
            this.stack2 = new Stack<>();
        }

        public <T> void enqueue(T value) {
            this.stack1.push(value);
        }

        public <T> void dequeue() {
            if (this.stack2.isEmpty()) {
                while (!this.stack1.isEmpty()) {
                    this.stack2.push(this.stack1.pop());
                }
            }
            this.stack2.pop();
        }

        public <T> Object peek() {
            if (this.stack2.isEmpty()) {
                while (!this.stack1.isEmpty()) {
                    this.stack2.push(this.stack1.pop());
                }
            }
            return this.stack2.peek();
        }

    }

    public static void main(String[] args) {
        MyQueue<Integer> queue = new MyQueue<Integer>();

        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();

        for (int i = 0; i < n; i++) {
            int operation = scan.nextInt();
            if (operation == 1) { // enqueue
              queue.enqueue(scan.nextInt());
            } else if (operation == 2) { // dequeue
              queue.dequeue();
            } else if (operation == 3) { // print/peek
              System.out.println(queue.peek());
            }
        }
        scan.close();
    }
}